internal/app/go-filecoin/node/node.go

```diff
@@ -679,12 +679,12 @@ func (node *Node) CreateMiningWorker(ctx context.Context) (*mining.DefaultWorker
 		TicketGen:      consensus.NewTicketMachine(node.PorcelainAPI),
 		TipSetMetadata: node.chain.ChainReader,
 
-		MessageSource: node.Messaging.Inbox.Pool(),
-		MessageStore:  node.chain.MessageStore,
-		Processor:     node.Chain().Processor,
-		Blockstore:    node.Blockstore.Blockstore,
-		Clock:         node.ChainClock,
-		Poster:        node.StorageMining.PoStGenerator,
+		MessageSource:  node.Messaging.Inbox.Pool(),
+		MessageStore:   node.chain.MessageStore,
+		PenaltyChecker: consensus.NewMessagePenaltyChecker(node.Chain().State),
```

Comment:
I'd maybe say call it `MessageQualifier` or `MessageSelecter` or something. Scope will very likely expand in the future.